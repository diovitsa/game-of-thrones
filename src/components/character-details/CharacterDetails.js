import React from 'react';
import { Text, View } from 'react-native';

function CharacterDetails({ navigation: { state: { params: { character } } } }) {
  const titles = character.titles.map(title => {
    return <Text>{title} </Text>
  });

  return (
    <View>
      <Text>
        Name: {character.name}
      </Text>
      <Text>
        Culture: {character.culture}
      </Text>
      <Text>
        Gender: {character.gender}
      </Text>
      <Text>
        Born: {character.born || 'no info'}
      </Text>
      <Text>
        Titles: {titles}
      </Text>
    </View>
  );
}

export default CharacterDetails;