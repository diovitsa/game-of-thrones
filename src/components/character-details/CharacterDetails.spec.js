import React from 'react';
import CharacterDetails from './CharacterDetails';
import renderer from 'react-test-renderer'

describe('CharacterDetails', () => {
  let component;
  // let componentInstance;

  const defaultProps = {
    navigation: {
      state: {
        params: {
          character: {
            name: 'testName',
            culture: 'testCulture',
            gender: 'Male',
            titles: ['testTitle']
          }
        }
      }
    }
  };


  it('should render properly', () => {
    component = renderer.create(<CharacterDetails {...defaultProps}/>).toJSON();
    expect(component).toMatchSnapshot()
  });
});