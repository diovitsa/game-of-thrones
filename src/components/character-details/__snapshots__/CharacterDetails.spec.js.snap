// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetails should render properly 1`] = `
<View>
  <Text>
    Name: 
    testName
  </Text>
  <Text>
    Culture: 
    testCulture
  </Text>
  <Text>
    Gender: 
    Male
  </Text>
  <Text>
    Born: 
    no info
  </Text>
  <RCTScrollView
    ListHeaderComponent={
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Titles: 
      </Text>
    }
    ListHeaderComponentStyle={
      Object {
        "fontSize": 19,
        "fontWeight": "bold",
      }
    }
    data={
      Array [
        "testTitle",
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "fontSize": 19,
            "fontWeight": "bold",
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Titles: 
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text>
          testTitle
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
